### DISCUSSION TITLE
Meeting: Celo Wallet and (d)App Council #16 [October] 💬

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/celo-monorepo#9875] [Wallet and (d)App Council]
# Monthly Celo Wallet and (d)App Council

##  📆 Quick links 

- When: **Wednesday, October 12 (9am PDT/6pm CEST)**
- Where: **[Google Meet](https://meet.google.com/uwj-wfse-uzm)**
- Agenda: **comments**👇
- Previous: https://github.com/celo-org/celo-monorepo/discussions/9774
- Next: https://github.com/celo-org/celo-monorepo/discussions/9955
- Recording: [Google Drive](https://drive.google.com/file/d/1OxUG3lKMCh9zJnkqPT7MzrMGPusvNJW8/view?usp=sharing)

## 🤷‍♂️ What is this?

A monthly meeting for Celo **developers, founders,** **product** people (anyone really!) to come together and
- find consensus (coordinate),
- learn about needs, pain points, and desires
- present and collect technical feedback

## 🗒️  Agenda

- Comment with a short sentence describing what you'd like to discuss
- Don't worry, it can be small or big! 
- Every builder is welcome (and we are very easy going, I promise 😄)
- Upvote topics you're interested in

## Housekeeping

_You can safely ignore this!_

- [x] Send update calendar invite with Github link
- [x] Encourage additions to the agenda
- [x] Publicise on forum and Discord
- [x] RECORD meeting 🎥
- [x] Share recording below and on forum
- [x] Create next meeting (link to next meeting above)

**Comment by fbaud** [comment]
I'd like to discuss workflows of interactions between Dapps and wallets with the following specific cases as an example:
- sending multiple atomic transactions (see discussion https://github.com/celo-org/celo-monorepo/discussions/9907)
- sending a plain text to be signed by a remote wallet

**Reply by arthurgousset** [reply]
Notes: 

- @fbaud:
  - User can connect from any interface on web and simply accept in wallet.
  - User has to make multiple actions in Valora. Can be cumbersome for users.
  - Sort of "app connects to wallet" (within some upper bounds for security)
  - Query: How can this kind of workflow work?
- @jeanregisser: 
  - WalletConnect does one request at a time
  - Visually users need to see all transactions they are signing ("accepting")
  - What are the transactions, signatures your app is making? Can you combine them in one transaction?
- @fbaud: 
  - at this point it's ~5 transactions
  - some use cases are minting NFTs
  - some concerns with gas limit
  - we reconnect with WalletConnect every time the user connects
  - idea: within a connection, an app has permission to make a certain number of transactions
  - "A user has to sign 5 times, and they don't really know what they are signing"
  - thinking about app authentication
- human: 
  > the Node Finance SDK allows builders to do multiSignTxns, which signs and dispatches an array of transactions in the order given. We are working to integrate a transaction simulator to communicate these actions to users before executing. Caveat is our SDK is for React Native only (at the moment)
- @fbaud: 
  - We're recording a bunch of metadata on the blockchain
- @jmrossy: 
  - Could you do more in a single smart contract? 
- @fbaud: 
  - today, the way we'll fix this is by using a local browser account that pays gas on a user's behalf (needs to be topped up)

**Comment by arthurgousset** [comment]
Highlight **Ethersjs feedback** discussion: 
- https://github.com/celo-org/celo-monorepo/discussions/9922

Here is some more context copied from above: 

> 👉 **TLDR: Please share/upvote issues you have with Ethers.js on Celo.** 👈
> 
> Hi all, @aaronmgdr is investigating issues you face with [Ethers.js](https://www.npmjs.com/package/ethers).
> 
> To help you, we'd like learn about your issues and to get as much context as possible.
> 
> Here are some problems on our radar:
> 
> 1. re **gas estimation** (missing `gasLimit` field in RPC API) has been fixed by @piersy in this PR: [Improve RPC API compatibility for Ethers.js celo-blockchain#1957](https://github.com/celo-org/celo-blockchain/pull/1957)
> 2. re **paying gas with custom tokens** (missing `feeCurrency` support in Ethers.js). This is on @aaronmgdr's radar 📡
> 
> Here is some feedback we heard:
> 
> > For me (as a developer), the only things I care about is to connect/disconnect to a wallet, get the user address, signer and provider (so I know if on the wrong network) and be able to set fee currency. [...] I don't want anything else other than a generic wallet connector and be able to set fee transactions. @celo/connect is working well for setting fees.

**Reply by arthurgousset** [reply]
Notes: 

- Rossy
  - 2x packages (celo-ethers-wrapper and hyperlane/celo-ethers-provider)
  - `gasLimit` is really only problem here
- Dylan: 
  - Main issues: custom gas tokens on Celo
- Aaron: 
  - the other problem that is related here is WalletConnect that drops the specified `feeCurrency`
  - this problem exists across client libraries (ethers, web3, contractkit which uses web3)
- Jean: 
  - Wonder if we should let wallets decide on `feeCurrency`, dapps don't really care as long as gas is paid.
  - Maybe user can change feeCurrency when they accept the transaction
  - Rossy/Aaron: That's a good idea
- Rossy: 
  - possible vulnerability: wallet modifies fields, but seems reasonable
- Dylan: 
  - In Node SDK we simply want a developer to set a `provider`
  - Aaron: we simply need an ethers signer

**Comment by arthurgousset** [comment]
Highlight @aaronmgdr's work on **Rainbowkit support**:
- [rainbowkit-with-celo.vercel.app](https://rainbowkit-with-celo.vercel.app/)

<img src="https://user-images.githubusercontent.com/46296830/195329710-878bf87b-fe59-4aba-ae73-b77fb1493596.png" alt="rainbowkit" width="400"/>

Ask: 
- We'd love to hear what you think about the config and what should be improved 🙏

**Reply by arthurgousset** [reply]
Notes: 

- @aaronmgdr:
  - for people who are just starting, this is simply a config with useful Celo default settings.
- @jmrossy: 
  - used Rainbowkit in the past, nice/solid library.
  - clarifying: this doesn't provide Celo transaction parity
- @aaronmgdr: 
  - only scope here is configs (no transactions)
- @jmrossy: 
  - Rainbowkit is built around wagmi
  - Wagmi is great library for web hooks
  - this will be a great way to get a dapp started fast
- @jeanregisser: 
  - how will you release/publish this config?
- @aaronmgdr:
  - it's an npm package, simply import.
  - @jmrossy: this could simply be a Github Gist (to copy/paste)

**Comment by arthurgousset** [comment]
(Brief follow-up for visibility)

The ASv2 contracts are deployed on Baklava, Alfajores and Mainnet.  🥳 
The CGP to add these to the core contracts is currently in the voting stage: [celo.stake.id/#/proposal/71](https://celo.stake.id/#/proposal/71).

You can find the contracts at the following addresses (I’m working on getting them verified in the explorers):

**Mainnet**:

- [FederatedAttestations](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/identity/FederatedAttestations.sol):
  - `0x0aD5b1d0C25ecF6266Dd951403723B2687d6aff2` (proxy)
  - `0x76A4daaC43912A443f098D413DED2Cb7A153EA85` (implementation)
- [OdisPayments](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/identity/OdisPayments.sol):
  - `0xAE6B29f31B96e61DdDc792f45fDa4e4F0356D0CB` (proxy)
  - `0x9Ea5E9b9B48a72325D59B3EBA147F42b1b14BF78` (implementation)
- [Escrow](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/identity/Escrow.sol):
  - `0xf4Fa51472Ca8d72AF678975D9F8795A504E7ada5` (proxy)
  - `0xcC4E6caBe88EBb7FCCB40d862bf1C3a89f88e835` (implementation)

**Alfajores**:

- [FederatedAttestations](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/identity/FederatedAttestations.sol):
  - `0x70F9314aF173c246669cFb0EEe79F9Cfd9C34ee3` (proxy)
  - `0x926E88a280902Bfff5047693B9CeAfdb9F4d5095` (implementation)
- [OdisPayments](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/identity/OdisPayments.sol):
  - `0x645170cdB6B5c1bc80847bb728dBa56C50a20a49` (proxy)
  - `0x72eAC1F0518213Ad405560eFd3fB647FbDAdb703` (implementation)
- [Escrow](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/identity/Escrow.sol):
  - `0xb07E10c5837c282209c6B9B3DE0eDBeF16319a37` (proxy)
  - `0xa34117B48313dE0093d599720998415bAb5FD61d` (implementation)

We're currently doing beta testing here: 

- [asv2](https://github.com/celo-org/identity/tree/jesse/example-scripts/asv2)/[example-scripts](https://github.com/celo-org/identity/tree/jesse/example-scripts/asv2/example-scripts)

**Comment by aaronmgdr** [comment]
What are your compile targets? Is anyone supporting browsers or node versions that need es5 or is es6 ok? 

- https://github.com/celo-org/celo-monorepo/discussions/9908

**Reply by arthurgousset** [reply]
Notes: 

- There is a poll above. Please vote above 👆 

<img width="600" alt="image" src="https://user-images.githubusercontent.com/46296830/195401729-9e4f261e-fe67-4f85-ab3d-e7eff2b9a612.png">

**Reply by arthurgousset** [reply]
- Rossy: 
  - Totally fine with es6 (most people targeting es5 do that for Internet Explorer and they will need to do a bunch of other changes anyway)
- Frederic: 
  - Most of the libraries are still published with es5.
  - "when it's not broken, don't fix it"
- Rossy: 
  - I think react script can accept es6/7 libraries

**Comment by arthurgousset** [comment]
Evan: https://github.com/Ubeswap/swap-component new ubeswap swap module released this weekend FYI

**Reply by aaronmgdr** [reply]
Im wondering why this isn't published on npm under the @ubeswap namespace?

**Reply by arthurgousset** [reply]
Notes: 
- just a quick update (helps decentralise the Ubeswap front-end)
- powered by the minima router (sources liquidity from all Celo native DEXes)

**Comment by arthurgousset** [comment]
📽 Here is the [recording](https://drive.google.com/file/d/1OxUG3lKMCh9zJnkqPT7MzrMGPusvNJW8/view?usp=sharing) for anyone that missed the meeting.
